atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -213,47 +215,118 @@ def assemble_srpm_url(self, base_url, srpm_filename, sign_key=None):
+        remote_json_map[remote_source['dest']] = cachito_json_url
+
+        archive_found = False
+        json_found = False
+        all_archives = []
+
+        for archive in archives:
+            if archive['filename'] == REMOTE_SOURCE_TARBALL_FILENAME:
+                archive_found = True
+            elif archive['filename'] == REMOTE_SOURCE_JSON_FILENAME:
+                json_found = True
+            all_archives.append(archive['filename'])
+
+        if not (archive_found and json_found):
+            message = ', '.join(part for t, part in (
+                (archive_found, "remote source archive missing"),
+                (json_found, "remote source json missing")) if not t)
+            raise RuntimeError(message)
+
+        elif len(archives) > 2:
+            raise RuntimeError('There can be just one remote sources archive and one '
+                               'remote sources json, got: {}'.format(all_archives))
+
+        return remote_sources_urls, remote_json_map
+
+    def _process_multiple_remote_sources(self, koji_build, archives, remote_sources_path):
+        self.log.debug('remote_sources defined')
+        remote_sources_urls = []
+        remote_json_map = {}
+        remote_sources = koji_build['extra']['typeinfo']['remote-sources']
+        wrong_archives = False
+        all_archives = []
+
+        for remote_s in remote_sources:
+            remote_archive = None
+            remote_json = None
+
+            if len(remote_s['archives']) != 2:
+                self.log.error('remote source "%s" does not contain 2 archives, but "%s"',
+                               remote_s['name'], remote_s['archives'])
+                wrong_archives = True
+            else:
+                for archive in remote_s['archives']:
+                    if archive.endswith('.json'):
+                        remote_json = archive
+                    else:
+                        remote_archive = archive
+
+                if not remote_json:
+                    self.log.error('remote source json, for remote source "%s" not found '
+                                   'in archives "%s"', remote_s['name'], remote_s['archives'])
+                    wrong_archives = True
+                else:
+                    remote_source = {}
+                    remote_source['url'] = os.path.join(remote_sources_path, remote_archive)
+                    remote_source['dest'] = '-'.join([koji_build['nvr'], remote_archive])
+                    remote_sources_urls.append(remote_source)
+                    cachito_json_url = os.path.join(remote_sources_path, remote_json)
+                    remote_json_map[remote_source['dest']] = cachito_json_url
+                    all_archives.append(remote_archive)
+                    all_archives.append(remote_json)
+
+        if wrong_archives:
+            raise RuntimeError('Problems with archives in remote sources: {}'.
```

Comment:
Is it possible to have more details here re: "Problems"?

Reply:
those are logged as errors above

Reply:
ACK, thx